analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1290 @@
+                    Set(ByVal value As Integer)
+                    End Set
+                End Property
+
+                Public Sub M()
+                    {{access}}
+                End Sub
+            End Class
+            """;
+        var context = ArgumentContextVB(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(
+            m => m is { MethodKind: var kind, ContainingType: { } type } && type.Name == "C" && (isGetter ? kind == MethodKind.PropertyGet : kind == MethodKind.PropertySet),
+            (n, c) => true,
+            (_, _, _) => true,
+            p => p.Name == parameterName, (_, _) => true);
+        var result = MatchArgumentVB(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("""process.Modules[$$0]""")]
+    [DataRow("""process?.Modules[$$0]""")]
+    [DataRow("""process.Modules?[$$0]""")]
+    [DataRow("""process?.Modules?[$$0]""")]
+    [DataRow("""process.Modules[index: $$0]""")]
+    [DataRow("""process?.Modules?[index: $$0]""")]
+    public void Indexer_ModulesAccess(string modulesAccess)
+    {
+        var snippet = $$"""
+            public class Test
+            {
+                public void M(System.Diagnostics.Process process)
+                {
+                    _ = {{modulesAccess}};
+                }
+            }
+            """;
+        var context = ArgumentContextCS(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(m => m is { MethodKind: MethodKind.PropertyGet, ContainingType: { } type } && type.Name == "ProcessModuleCollection",
+            (n, c) => n.Equals("Modules", c), (_, _, _) => true, p => p.Name == "index", (_, p) => p is null or 0);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("""processStartInfo.Environment[$$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment[$$"TEMP"]""")]
+    [DataRow("""processStartInfo.Environment?[$$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment?[$$"TEMP"]""")]
+    [DataRow("""processStartInfo.Environment[key: $$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment?[key: $$"TEMP"]""")]
+    public void Indexer_Environment(string environmentAccess)
+    {
+        var snippet = $$"""
+            public class Test
+            {
+                public void M(System.Diagnostics.ProcessStartInfo processStartInfo)
+                {
+                    _ = {{environmentAccess}};
+                }
+            }
+            """;
+        var context = ArgumentContextCS(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(KnownType.System_Collections_Generic_IDictionary_TKey_TValue, "Environment", p => p.Name == "key", 0);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+#if NET5_0_OR_GREATER
+
+    [TestMethod]
+    public void Attribute_Obsolete()
```

Comment:
What is special about the obsolete attribute? I think the test name should reflect the functionality tested and not the Type name which can be anything. Do I miss something?

Reply:
There is nothing special about it. The name of the tested symbol is often given in the test names, e.g., `Indexer_ModulesAccess.`

Reply:
Yes, I've also noticed `Modules.` And I would rename that too.

Wouldn't be better to specify the tested functionality of the descriptor instead of the Type name?